pkg/controller/dnsendpoint/common.go

```diff
@@ -118,8 +118,14 @@ func DedupeAndMergeEndpoints(endpoints []*feddnsv1a1.Endpoint) (result []*feddns
 	for i := 1; i < len(endpoints); {
 		if endpoints[i].DNSName == endpoints[i-1].DNSName {
 			// Merge targets
-			endpoints[i-1].Targets = append(endpoints[i-1].Targets, endpoints[i].Targets...)
-			endpoints[i-1].Targets = sortAndRemoveDuplicateTargets(endpoints[i-1].Targets)
+			if endpoints[i-1].RecordType != endpoints[i].RecordType {
+				if endpoints[i-1].RecordType == RecordTypeCNAME {
```

Comment:
(Suggestion) Better to add a comment on why we are doing this, which will help future maintainers.

Reply:
done~

Reply:
s/enpoint/endpoint/
s/at same region or at same az/at same region and az/
s/if cluster no endpoints/if cluster does not have endpoints/

Reply:
thanks. 

fixed. ^^